layout/layout.go

```diff
@@ -0,0 +1,740 @@
+	return func(i *imageOptions) error {
+		i.platform = platform
+		return nil
+	}
+}
+
+// WithCreatedAt lets a caller set the created at timestamp for the image.
+// Defaults for a new image is imgutil.NormalizedDateTime
+func WithCreatedAt(createdAt time.Time) ImageOption {
+	return func(i *imageOptions) error {
+		i.createdAt = createdAt
+		return nil
+	}
+}
+
+// FromBaseImagePath loads an existing image as the config and layers for the new underlyingImage.
+// Ignored if underlyingImage is not found.
+func FromBaseImagePath(path string) ImageOption {
+	return func(i *imageOptions) error {
+		i.baseImagePath = path
+		return nil
+	}
+}
+
+func NewImage(path string, ops ...ImageOption) (*Image, error) {
+	imageOpts := &imageOptions{}
+	for _, op := range ops {
+		if err := op(imageOpts); err != nil {
+			return nil, err
+		}
+	}
+
+	platform := defaultPlatform()
+	if (imageOpts.platform != imgutil.Platform{}) {
+		platform = imageOpts.platform
+	}
+
+	image, err := emptyImage(platform)
+	if err != nil {
+		return nil, err
+	}
+
+	ri := &Image{
+		Image: image,
+		path:  path,
+	}
+
+	if imageOpts.prevImagePath != "" {
+		if err := processPreviousImageOption(ri, imageOpts.prevImagePath, platform); err != nil {
+			return nil, err
+		}
+	}
+
+	if imageOpts.baseImagePath != "" {
+		if err := processBaseImageOption(ri, imageOpts.baseImagePath, platform); err != nil {
+			return nil, err
+		}
+	} else if imageOpts.baseImage != nil {
+		ri.Image = imageOpts.baseImage
+	}
+
+	if imageOpts.createdAt.IsZero() {
+		ri.createdAt = imgutil.NormalizedDateTime
+	} else {
+		ri.createdAt = imageOpts.createdAt
+	}
+
+	return ri, nil
+}
+
+func processPreviousImageOption(ri *Image, prevImagePath string, platform imgutil.Platform) error {
+	prevImage, err := newV1Image(prevImagePath, platform)
+	if err != nil {
+		return err
+	}
+
+	prevLayers, err := prevImage.Layers()
+	if err != nil {
+		return errors.Wrapf(err, "getting layers for previous image with path %q", prevImagePath)
+	}
+
+	ri.prevLayers = prevLayers
+
+	return nil
+}
+
+func processBaseImageOption(ri *Image, baseImagePath string, platform imgutil.Platform) error {
+	baseImage, err := newV1Image(baseImagePath, platform)
+	if err != nil {
+		return err
+	}
+
+	ri.Image = baseImage
+
+	return nil
+}
+
+func emptyImage(platform imgutil.Platform) (v1.Image, error) {
+	cfg := &v1.ConfigFile{
+		Architecture: platform.Architecture,
+		OS:           platform.OS,
+		OSVersion:    platform.OSVersion,
+		RootFS: v1.RootFS{
+			Type:    "layers",
+			DiffIDs: []v1.Hash{},
+		},
+	}
+	return mutate.ConfigFile(empty.Image, cfg)
+}
+
+func defaultPlatform() imgutil.Platform {
+	return imgutil.Platform{
+		OS:           "linux",
+		Architecture: "amd64",
+	}
+}
+
+func (i *Image) Label(key string) (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
```

Comment:
```suggestion
		return "", fmt.Errorf(err, "getting config for image at path %q: %w", i.path, err)
```
(to be consistent with following error and remove use of deprecated `github.com/pkg/errors`)

Reply:
Since this sort of error handling is used a lot in this file, maybe we could make a helper for it